client/state/sharing/publicize/publicize-actions/reducer.js

```diff
@@ -53,6 +57,11 @@ export const scheduled = createReducer( {}, {
 		),
 	[ PUBLICIZE_SHARE_ACTION_EDIT_SUCCESS ]:
 		( state, { siteId, postId, item } ) => updateDataForPost( item, state, siteId, postId, item.ID ),
+	[ PUBLICIZE_SHARE_ACTION_SCHEDULE_SUCCESS ]:
+		( state, { siteId, postId, items } ) => {
+			items.forEach( item => state = updateDataForPost( item, state, siteId, postId, item.ID ) );
```

Comment:
Usually usages of `forEach` can be done more clearly with `map` or `reduce`. Could we do this in a different way that doesn't mutate `state` for each iteration?

I think I understand why you did it this way, though, because `updateDataForPost` appears to make some assumptions about changing just one item. It might be that `updateDataForPost` needs to be simplified so it can be used more easily here. I'm happy to help do that if you can help me understand what this is supposed to do.

Adding some tests for these actions would also help clarify how this is supposed to work and verify that it does.